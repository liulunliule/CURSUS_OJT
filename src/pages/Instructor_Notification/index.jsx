import React, { useEffect, useState } from 'react';
import "./index.scss";
import Bell from "../../assets/img/bell.png";
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchInstructorNotifications } from '../../redux/features/instructorNotificationsSlice';

const Instructor_Notification = () => {
    const [mouseOverColor, setMouseOverColor] = useState(null);

    const mouseOverOn = () => {
        setMouseOverColor("black");
    }

    const mouseOverOff = () => {
        setMouseOverColor(null);
    }

    const dispatch = useDispatch();
    const notification = useSelector((state) => state.instructorNotifications.instructorNotifications)
    const loading = useSelector((state) => state.instructorNotifications.loading);

    useEffect(() => {
        dispatch(fetchInstructorNotifications());
    }, [dispatch])

    const isShowAll = useSelector((state) => state.savedCourse.isShowAll);

  return (
    <div className={`wrapper12 ${isShowAll ? "active" : ""}`}>
        <div className='sa4d25'>
            <div className='container-fluid'>
                <div className='row'>
                    <div className='col-lg-12'>
                        <h2 className='noti-title'>
                            <img src={Bell} className=''/>
                            Notifications
                        </h2>
                    </div>
                </div>

                <div className='row'>
                    <div className='setting-section mbt-30'>
                        <Link to="/setting_page/account_tab">
                        <button
                                style={{backgroundColor: mouseOverColor, marginBottom: 30}}
                                class="setting_noti"
                                onMouseOver={mouseOverOn}
                                onMouseOut={mouseOverOff}
                                >
                                Notification Setting
                                </button>
                        </Link>
                               
                                    {notification.map((d) => (
                                        <div className='message_back'>
                                            <div className='panel_stuff'>
                                                <div className='profile_show'>
                                                    <img src={d.img} alt='photo'/>
                                                    <div className='content_show'>
                                                        <h6>{d.name}</h6>
                                                        <p>{d.content} <strong>{d.strong}</strong></p>
                                                        <span className='time_ago'>{d.time}</span>
                                                    </div>
                                                </div>
                                            </div>
                                        </div>
                                    ))}
                    </div>
                </div>
            </div>
        </div>
    </div>
  )
}


export default Instructor_Notification