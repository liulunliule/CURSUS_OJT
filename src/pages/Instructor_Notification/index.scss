.noti-title {
    font-size: 20px;
    margin-top: 0 !important;
    margin-bottom: 8px !important;
    font-weight: 500;
    font-family: "Roboto", sans-serif;
    color: #333;
    margin-left: 25px;

    img {
        width: 20px;
        height: 20px;
        margin-right: 10px;
    }
}

.wrapper12.active {
    transition: margin-left 0.5s ease;
    .sa4d25_sa {
        margin-left: 60px;
    }
}

.setting-section {
    flex: 0 0 auto;
    width: 100%;
}

.setting_noti {
    margin-top: 40px;
    float: left;
    padding: 10px 20px;
    color: #fff;
    background: #ed2a26;
    font-size: 14px;
    font-weight: 500;
    font-family: "Roboto", sans-serif;
    border: none;
    margin-left: 25px;
}

.message_back {
    background: #fff;
    border-radius: 3px;
    border: 1px solid #efefef;
    padding: 0;
    float: left;
    width: 95%;
    margin-left: 25px;

    &:hover {
        background-color: #ffecec;
    }
}

.panel_stuff {
    border-bottom: 1px solid #efefef !important;
    min-width: 100%;
    max-width: 100%;
    white-space: normal;
    display: block;
    padding: 20px !important;
}

.profile_show {
    display: flex;

    img {
        width: 40px;
        height: 40px;
        margin-right: 10px;
        border: 2px solid #fff;
        border-radius: 100%;
        box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
    }
}

.content_show {
    width: 80%;
    white-space: normal;

    h6 {
        font-size: 14px;
        font-weight: 500;
        font-family: "Roboto", sans-serif;
        margin-bottom: 5px;
        color: #333;
        text-align: left;
        position: relative;
    }

    p {
        font-size: 14px;
        font-weight: 400;
        font-family: "Roboto", sans-serif;
        margin-bottom: 8px;
        color: #686f7a;
        text-align: left;
        margin-top: 8px;
        line-height: 20px;
    }
}

.time_ago {
    font-size: 12px;
    font-weight: 400;
    font-family: "Roboto", sans-serif;
    margin-bottom: 0;
    color: #afafaf;
    text-align: left;
}

.mart-30 {
    margin-top: 30px;
}

.mbt-30 {
    margin-bottom: 30px;
}
