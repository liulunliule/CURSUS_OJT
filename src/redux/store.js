import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
