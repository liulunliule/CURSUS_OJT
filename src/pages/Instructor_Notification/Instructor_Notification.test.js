import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Instructor_Notification from '.';

test('test verification page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Instructor_Notification/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});