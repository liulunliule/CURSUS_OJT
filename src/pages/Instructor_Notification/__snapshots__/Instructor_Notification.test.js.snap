// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test verification page 1`] = `
<div
  className="wrapper12 "
>
  <div
    className="sa4d25"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h2
            className="noti-title"
          >
            <img
              className=""
              src="bell.png"
            />
            Notifications
          </h2>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="setting-section mbt-30"
        >
          <a
            href="/setting_page/account_tab"
            onClick={[Function]}
          >
            <button
              class="setting_noti"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "backgroundColor": null,
                  "marginBottom": 30,
                }
              }
            >
              Notification Setting
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
